src/lib.rs

```diff
@@ -477,23 +648,115 @@ pub struct Mesh<'a> {
 impl<'a> Mesh<'a> {
     /// Creates a new [`Mesh`] object to be partitioned.
     ///
+    /// `nparts` is the number of parts wanted in the mesh partition.
+    ///
+    /// # Input format
+    ///
+    /// The length of `eptr` is `n + 1`, where `n` is the number of elements in
+    /// the mesh. The length of `eind` is the sum of the number of nodes in all
+    /// the elements of the mesh. The list of nodes belonging to the `i`th
+    /// element of the mesh are stored in consecutive locations of `eind`
+    /// starting at position `eptr[i]` up to (but not including) position
+    /// `eptr[i+1]`.
+    ///
+    /// # Errors
+    ///
+    /// The following invariants must be held, otherwise this function returns
+    /// an error:
+    ///
+    /// - `nparts` is strictly greater than zero,
+    /// - `eptr` has at least one element (its length is the one more than the
+    ///   number of mesh elements),
+    /// - `eptr` is sorted,
+    /// - elements of `eptr` are positive,
+    /// - the last element of `eptr` is the length of `eind`,
+    /// - all the arrays have a length that can be held by an [`Idx`].
+    ///
+    /// # Mutability
+    ///
+    /// [`Mesh::part_dual`] and [`Mesh::part_nodal`] may mutate the contents of
+    /// `eptr` and `eind`, but should revert all changes before returning.
+    pub fn new(
+        nparts: Idx,
+        eptr: &'a mut [Idx],
+        eind: &'a mut [Idx],
+    ) -> std::result::Result<Mesh<'a>, NewError> {
+        if nparts <= 0 {
```

Comment:
```suggestion
        if nparts <= 1 {
```
I'm not sure if it's ok if `nparts == 1`. The documentation doesn't say anything, but this test fails with `SIGFPE`:
```rust
// using old version
#[test]
fn mesh_sigfpe() {
    let eptr = &mut [0, 3, 6];
    let eind = &mut [0, 1, 2, 1, 2, 3];

    let mut epart = [0; 2];
    let mut npart = [0; 4];


    // passing npart == 1 raises SIGFPE
    Mesh::new(4, 1, eptr, eind)
        .part_dual(&mut epart, &mut npart)
        .unwrap();
}
```

There is [behavior in `METIS_PartGraphKway`][0] for handling `nparts == 1` if debug level 512 is set, but that [doesn't even have a declaration in `metis.h`][1].

[0]: https://github.com/KarypisLab/METIS/blob/e0f1b88b8efcb24ffa0ec55eabb78fbe61e58ae7/libmetis/kmetis.c#L72
[1]: https://github.com/KarypisLab/METIS/blob/e0f1b88b8efcb24ffa0ec55eabb78fbe61e58ae7/include/metis.h#L337

Reply:
nice.. there is also this log2 into div by 0 that unconditionally triggers the floating point exception:

https://github.com/KarypisLab/METIS/blob/e0f1b88b8efcb24ffa0ec55eabb78fbe61e58ae7/libmetis/kmetis.c#L56

We can handle the `nparts==1` on the rust side I guess.